packages/flutter/test/painting/matrix_utils_test.dart

```diff
@@ -21,6 +21,11 @@ void main() {
     expect(MatrixUtils.getAsTranslation(test), isNull);
     test = new Matrix4.translationValues(1.0, 2.0, 0.0);
     expect(MatrixUtils.getAsTranslation(test), equals(const Offset(1.0, 2.0)));
+    // Rotate matrix and then rotate it back to introduce tiny rounding errors.
+    test.rotateZ(3.0);
+    test.rotateZ(-3.0);
```

Comment:
i think we should be discouraging this kind of practice, rather than papering over the errors it introduces

Reply:
Very reasonable looking code in the gallery sample app generates rounding errors in matrix transform that makes a scale transform not look like a scale.  Obviously this rotate example is egregious to generate a simple repro to but floating point rounding error is a real problem for this code detecting that matrices represent a scale.

Reply:
Can you point me to the offending code or file a bug?

We suspect that we have some weird rounding error in a rotation somewhere and it's causing actual visible artifacts (visible e.g. in the dashboards on the second floor of my building), so I am very interested in anything you might have found that resembles that kind of problem.

Reply:
I've looked around and I can't find a case where rounding errors are showing up in practice in the Gallery application. I distinctly remember it back in the Dart Conf time frame but I may have just been getting confused by some scale transforms that weren't showing up as a simple scales because they also scaled Z as well as X and Y.
Are you alright with transforms such as
      transform: new Matrix4.identity()..scale(2.0),

showing up as
transform: scale(2.0)
even though technically they are scaling the z axis as well?
I could also show them as
transform: scale(2.0, 2.0, 2.0) to be precise about the meaning.
Let me know.

Reply:
Yeah showing `scale(2.0)` is totally fine.